de.fhdo.lemma.eclipse.ui/src/de/fhdo/lemma/eclipse/ui/specify_url_dialog/SpecifyUrlDialog.xtend

```diff
@@ -0,0 +1,274 @@
+        dataDataModelName.grabExcessHorizontalSpace = true
+        dataDataModelName.horizontalAlignment = GridData.FILL
+        dataDataModelName.horizontalSpan = 3
+
+        txtDataModelName = new Text(container, SWT.SINGLE)
+        txtDataModelName.message = "MyServiceData"
+        txtDataModelName.setLayoutData(dataDataModelName)
+    }
+
+    private def createServiceModelName(Composite container) {
+        val lblServiceModelName = new Label(container, SWT.NONE)
+        lblServiceModelName.setText("Service Model Name: ")
+
+        val dataServiceModelName = new GridData()
+        dataServiceModelName.grabExcessHorizontalSpace = true
+        dataServiceModelName.horizontalAlignment = GridData.FILL
+        dataServiceModelName.horizontalSpan = 3
+
+        txtServiceModelName = new Text(container, SWT.SINGLE)
+        txtServiceModelName.message = "MyServiceService"
+        txtServiceModelName.setLayoutData(dataServiceModelName)
+    }
+    private def createServicePrefix(Composite container) {
+        val lblServicePrefix = new Label(container, SWT.NONE)
+        lblServicePrefix.setText("Service Model Prefix: ")
+
+        val dataServicePrefix = new GridData()
+        dataServicePrefix.grabExcessHorizontalSpace = true
+        dataServicePrefix.horizontalAlignment = GridData.FILL
+        dataServicePrefix.horizontalSpan = 3
+
+        txtServicePrefix = new Text(container, SWT.SINGLE)
+        txtServicePrefix.message = "de.example"
+        txtServicePrefix.setLayoutData(dataServicePrefix)
+    }
+
+    private def createTechnologyModelName(Composite container) {
+        val lblTechnologyModelName = new Label(container, SWT.NONE)
+        lblTechnologyModelName.setText("Technology Model Name: ")
+
+        val dataTechnologyModelName = new GridData()
+        dataTechnologyModelName.grabExcessHorizontalSpace = true
+        dataTechnologyModelName.horizontalAlignment = GridData.FILL
+        dataTechnologyModelName.horizontalSpan = 3
+
+        txtTechnologyModelName = new Text(container, SWT.SINGLE)
+        txtTechnologyModelName.message = "OpenApi"
+        txtTechnologyModelName.setLayoutData(dataTechnologyModelName)
+    }
+
+    /**
+     * Stores the textfield values to actual fields.
+     * Is called when a button in the dialog is pressed.
+     */
+    private def saveInput() {
+        this.fetchUrl = try {
+            new URL(txtUrl.getText)
+        } catch (MalformedURLException e) {
+            MessageDialog.openError(this.shell, "Invalid URL",
+                '''«txtUrl.getText» is not a valid url!'''
+            )
+            return false
+        }
+            this.targetLocation = txtTargetLocation.getText
+            this.dataName = txtDataModelName.getText
+            this.serviceName = txtServiceModelName.getText
+            this.technologyName = txtTechnologyModelName.text
+        if(!this.targetLocation.trim.isEmpty && !this.dataName.trim.isEmpty &&
```

Comment:
To increase readability...
- remove `this`
- introduce space between `if` and `(`
- introduce a helper method, e.g., `isBlankOrEmpty()`, for `trim.isEmpty()`
- regroup the predicates:
```
if (!targetLocation.isBlankOrEmpty() &&
    !dataName.isBlankOrEmpty() &&
    ...)
```